Doxysphinx docs

Maetveis opened the pull request:
Replaces: #3171

~~Requires https://github.com/RadeonOpenCompute/rocm-docs-core/pull/36~~ (merged)

cgmb committed:
Reorganize documenation based on rocm-docs-core

Commit:
Move build, install and contributing guides to docs

saadrahim commented:
@Maetveis the RTD build is not passing. Please help solve that.

saadrahim commented:
I suspect the builds are failing as you separated out the doxygen fixes. Perhaps it was wrong advice to suggest that.

Commit:
Bump readthedocs config to Ubuntu 20.04

Commit:
Add more direct links to the API modules

Commit:
Pin rocm-docs-core version, add dependabot

- Use the pip package to have rocm-docs-core pinned
- Add dependabot config to update pinned packages (including rocm-docs-core)

saadrahim commented on build-docs.sh:
Should this be removed? I don't think it is necessary.

Maetveis replied:
Agreed, it was part of the original PR that I took over, removed.

saadrahim commented on docs/.doxygen/Doxyfile:
```diff
@@ -32,19 +32,19 @@ DOXYFILE_ENCODING      = UTF-8
 # control system is used.
 
-PROJECT_NUMBER         =
+PROJECT_NUMBER         = v3.0.1.0
```
Why is there a version number here?

Maetveis replied:
Removed now. I reviewed the changes to the doxyfile to only change relevant values.

Commit:
fixup! Add more direct links to the API modules
